packages/insight-previous/src/providers/price/price.ts

```diff
@@ -39,8 +37,8 @@ export class PriceProvider {
         },
         err => {
           this.currencyProvider.loading = false;
-          this.logger.error(err);
           this.showErrorToast();
+          throw err;
```

Comment:
I don't recommend to show an error ("this.showErrorToast") and then `throw err`. Providers shouldn't send message directly to the user. 
Keep only `throw err` and let the component or page to make the message to the user